libcontainer/seccomp/seccomp_linux.go

```diff
@@ -170,29 +170,55 @@ func matchCall(filter *libseccomp.ScmpFilter, call *configs.Syscall) error {
 	// Convert the call's action to the libseccomp equivalent
 	callAct, err := getAction(call.Action)
 	if err != nil {
-		return err
+		return fmt.Errorf("action in seccomp profile is invalid: %s", err)
 	}
 
 	// Unconditional match - just add the rule
 	if len(call.Args) == 0 {
 		if err = filter.AddRule(callNum, callAct); err != nil {
-			return err
+			return fmt.Errorf("error adding seccomp filter rule for syscall %s: %s", call.Name, err)
 		}
 	} else {
-		// Conditional match - convert the per-arg rules into library format
+		// If two or more arguments have the same condition,
+		// Revert to old behavior, adding each condition as a separate rule
+		argCounts := make([]uint, 6)
 		conditions := []libseccomp.ScmpCondition{}
 
 		for _, cond := range call.Args {
+			argCounts[cond.Index] += 1
```

Comment:
And I'm worried this will panic if someone gives an invalid index as a configuration. We should error out here if `cond.Index >= MaxSyscallArguments`.

Reply:
I think we can actually just move the line down below the logic that creates a Libseccomp condition, which has logic to catch invalid indexes. Should resolve the issue without introducing additional checks.